docs/_docs/rql.md

```diff
@@ -0,0 +1,470 @@
+---
+title: Resource Query Language
+---
+
+* [Background](#background)
+* [AST Grammar](#ast-grammar)
+* [Primaries](#primaries)
+  * [action](#action)
+  * [boolean](#boolean)
+  * [name](#name)
+  * [cname](#cname)
+  * [path](#path)
+  * [kind](#kind)
+  * [atime](#atime)
+  * [crtime](#crtime)
+  * [ctime](#ctime)
+  * [mtime](#mtime)
+  * [size](#size)
+  * [meta](#meta)
+* [Detailed Meta Primary Overview](#detailed-meta-primary-overview)
+
+## Background
+
+The resource query language (RQL) lets you query Wash resources. You can submit your queries to the API's `find` endpoint. Here's an example query
+
+```
+Eniss-MacBook-Pro:wash enisinan$ curl -X POST --unix-socket /tmp/WASH_SOCKET --header "Content-Type: application/json" --data '["kind", ["glob", "*ec2*instance"]]' 'http://localhost:/fs/find?path=/tmp/WASH_MOUNT/aws/wash' 2>/dev/null | jq
+[
+  {
+    "type_id": "aws::github.com/puppetlabs/wash/plugin/aws/ec2Instance",
+    "path": "/tmp/WASH_MOUNT/aws/wash/resources/ec2/instances/i-04621c13583930e6c",
+...
+```
+
+> Note that I started my own Wash server instance with `WASH_SOCKET="/tmp/WASH_SOCKET" ./wash server /tmp/WASH_MOUNT`
+
+This query returns all entries under the `aws/wash` entry whose `kind` matches the glob `*ec2*instance`. Informally, this query returns all AWS EC2 instances under the `wash` profile.
+
+You can view the [API docs]({{'/docs/api' | relative_url}}) for more details on the `find` endpoint, including its query parameters (not to be confused with an RQL query, which is specified in the request body).
+
+## AST Grammar
+
+This section documents the RQL's AST grammar. For convenience, let `PE <PredicateType>` denote the following grammar (where `PE` => `PredicateExpression`).
+
+```
+PE <PredicateType> :=
+  [BinaryOp, PE <PredicateType>, PE <PredicateType>] |
+  <PredicateType>
+
+BinaryOp := "AND" | "OR"
+```
+
+where `<PredicateType>` has its own AST grammar. Informally, `PE <PredicateType>` translates to `a predicate expression of <PredicateType>s`. For example, `PE Primary` translates to `a predicate expression of primaries`.
+
+Similarly, let `NPE <PredicateType>` denote the following grammar (where `NPE` => `NegatablePredicateExpression`):
+
+```
+NPE <PredicateType> :=
+  [UnaryOp,  NPE <PredicateType>]                       |
```

Comment:
Since it's already specialized with "negatable", maybe drop the generic Unary and just inline `'NOT'`.